CONTRIBUTING.md

```diff
@@ -1,6 +1,96 @@
+When preparing a contribution we ask that you read the developers guide (see `dev_guide/`) to check
+for design and coding style intentions.
+As described in the build tools documentation (`build_tools/build_system_readme.md`) the `flint`
+program is used to apply linting rules to the ASiMoV-CCS source.
 
-Before a contribution is merged it will be reviewed by one of the ASiMoV-CCS developers, in particular to check the coding standard is followed and to verify the full test suite must still pass.
+Before a contribution is merged it will be reviewed by one of the ASiMoV-CCS developers, in
+particular to check the coding standard is followed and to verify the full test suite must still
+pass.
+
+# Instructions for contributing on GitHub
+
+If you have not done so already, create a fork of the ASiMoV-CCS repository where you can prepare
+your contributions.
+
+We request that contributions are based on, and target, the `upstreaming` branch - this will help us
+to upstream the contributions.
+The `upstreaming` branch should be kept up to date with the `develop` branch to include the latest
+developments.
+
+## Before starting a contribution
+
+We appreciate people taking the time and effort to contribute to ASiMoV-CCS.
+To prevent potential duplication of effort, we ask that before embarking on making code changes that
+you open an issue on the GitHub repository outlining the proposed change and why it is necessary, it
+may be that someone is already working on something similar or that there would be a conflict with
+some other planned developments, this will give the opportunity to flag potential issues in advance.
+
+## Create a new branch
+
+Hopefully there are no clashes with the proposed contribution and ongoing work and the proposed work
+can proceed.
+In the issue proposing the contribution click the `Create a branch` link, this will open a popup
+where you can name your branch and select your forked repository as the branch destination.
+You should also click `Change branch source` and select `upstreaming` as the basis for your branch.
+At this point you will be able to checkout your branch locally and begin working.
+
+### Before you start
+
+Although the ASiMoV-CCS developers will try to keep `upstreamin` synchronised with `develop` it is
+worth ensuring your working branch is up to date before beginning development.
+To do so we recommend using `git rebase` to keep the git history clean, assuming that the main
+ASiMoV-CCS repository is `origin` and your branch is called `my-branch`:
+```
+git checkout develop
+git pull origin/develop
+git checkout my-branch
+git rebase develop
+```
+
+You are now ready to begin.
+
+### Keeping up to date
+
+We recommend periodically pulling the `origin/develop` branch and merging into your working branch
+if the development of your contribution continues for some time, this will ensure that it does not
+stray too far from the eventual merge target.
```

Comment:
Is this a merge instead of a rebase on purpose?

Reply:
Done

Reply:
Yep.